gk/main.c

```diff
@@ -96,6 +487,13 @@ run_gk(struct gk_config *gk_conf)
 	if (!gk_conf)
 		return -1;
 
+	gk_conf->num_lcores = gk_conf->lcore_end_id - gk_conf->lcore_start_id + 1;
+	gk_conf->instances = (struct gk_instance *)rte_calloc(NULL,
+		gk_conf->num_lcores, sizeof(struct gk_instance), 0);
```

Comment:
If any of the calls to `rte_eal_remote_launch()` fails below, then `gk_conf->instances` should be freed. I think the setup of each instance should actually be done in the `for` loop below, as described in another review comment.